### ISSUE TITLE
parser: `?` after a type should produce a parse error

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5982]
This snippet should produce a parse error, but it instead produces a `TSJsdocType`
* [Oxc Playground](https://playground.oxc.rs/#eNpFjtEKwyAMRX9F8rRBv0Do9rb/sG0cgiYl2q5S/Pcpw+4p9yTh3nvCDBpSXlEdalQ3y6wVbWFCed7V+FA7uwUGYNAnyEZtxEzJHKCTbDiAd5S6jjOveEEOE/tOSQxFyxJAW+MjlgFWIxGlOlbdXLq+Xvti5gXf2LIrBEfOuv+JkrB/ef60ajvKxLFW+GWU8gUIDU3q)
* [TypeScript Playground](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAHlC8UAUAzA9mgXFAdgVwFsAjCAJwH4BKBAPigDc0BLAEyA)
```ts
type x = (foo: number?) => void
```

**Comment by IWANABETHATGUY** [comment]
this case is also invalid, 
```
type x = (foo: number!) => void
```
[ts](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAHlC8UAUAzA9mgXFAdgVwFsAjCAJwEIBKBAPigDc0BLAEwChRIoBBBXQkqQDcHcNAAqfANowA-Nm4BdIUA)
[oxc](https://playground.oxc.rs/#eNpFjkEOgzAMBL+S+tRKvCASPfYVvQRwqkiJjZxAQSh/b6IqcPKObe3uASNoSPuMalO9ultmrWgJA8rtofqnWtlNb4IOGPQBslAdcadkNtBJFuzAO0pNx5FnPGEPA/tGSQxFyxJAW+Mj5g5mIxGlOBZdXZo+X9ti5Ak/WLMLBEfOuutESdi/PH9rtRVl4Fgq/DNy/gGcY06W)

**Comment by Boshen** [comment]
The AST is correct. Expect we need to produce an error:

```
    function checkJSDocTypeIsInJsFile(node: Node): void {
        if (!isInJSFile(node)) {
            if (isJSDocNonNullableType(node) || isJSDocNullableType(node)) {
                const token = tokenToString(isJSDocNonNullableType(node) ? SyntaxKind.ExclamationToken : SyntaxKind.QuestionToken);
                const diagnostic = node.postfix
                    ? Diagnostics._0_at_the_end_of_a_type_is_not_valid_TypeScript_syntax_Did_you_mean_to_write_1
                    : Diagnostics._0_at_the_start_of_a_type_is_not_valid_TypeScript_syntax_Did_you_mean_to_write_1;
                const typeNode = node.type;
                const type = getTypeFromTypeNode(typeNode);
                grammarErrorOnNode(
                    node,
                    diagnostic,
                    token,
                    typeToString(
                        isJSDocNullableType(node) && !(type === neverType || type === voidType)
                            ? getUnionType(append([type, undefinedType], node.postfix ? undefined : nullType)) : type,
                    ),
                );
            }
            else {
                grammarErrorOnNode(node, Diagnostics.JSDoc_types_can_only_be_used_inside_documentation_comments);
            }
        }
    }
```
